packages/jellyfish-api-core/__tests__/category/loan/depositToVault.test.ts

```diff
@@ -0,0 +1,216 @@
+      ownerAddress: await tGroup.get(0).generateAddress(),
+      loanSchemeId: 'scheme'
+    })
+    await tGroup.get(0).generate(1)
+    await tGroup.waitForSync()
+
+    vaultId1 = await tGroup.get(0).rpc.loan.createVault({
+      ownerAddress: await tGroup.get(0).generateAddress(),
+      loanSchemeId: 'scheme'
+    })
+    await tGroup.get(0).generate(1)
+    await tGroup.waitForSync()
+  }
+
+  it('should be failed as first deposit must be DFI', async () => {
+    const promise = tGroup.get(0).rpc.loan.depositToVault({
+      id: vaultId, from: collateralAddress, amount: '1@BTC'
+    })
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow('At least 50% of the vault must be in DFI thus first deposit must be DFI')
+  })
+
+  it('should be failed as insufficient fund', async () => {
+    const promise = tGroup.get(0).rpc.loan.depositToVault({
+      id: vaultId, from: collateralAddress, amount: '99999@DFI'
+    })
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow(`Insufficient funds: can't subtract balance of ${collateralAddress}: amount 20000.00000000 is less than 99999.00000000`)
+  })
+
+  it('should be failed as different auth address', async () => {
+    const addr = await tGroup.get(1).generateAddress()
+    const promise = tGroup.get(0).rpc.loan.depositToVault({
+      id: vaultId, from: addr, amount: '1@DFI'
+    })
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow(`Incorrect authorization for ${addr}`)
+  })
+
+  it('should be failed as vault is not exists', async () => {
+    const promise = tGroup.get(0).rpc.loan.depositToVault({
+      id: '0'.repeat(64), from: collateralAddress, amount: '10000@DFI'
+    })
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow(`Vault <${'0'.repeat(64)}> not found`)
+  })
+
+  it('should depositToVault', async () => {
+    {
+      const vaultBefore = await tGroup.get(0).container.call('getvault', [vaultId])
+      const vaultBeforeDFIAcc = vaultBefore.collateralAmounts.length > 0
+        ? vaultBefore.collateralAmounts.find((amt: string) => amt.split('@')[1] === 'DFI')
+        : undefined
+      const vaultBeforeDFIAmt = vaultBeforeDFIAcc !== undefined ? Number(vaultBeforeDFIAcc.split('@')[0]) : 0
+
+      const depositId = await tGroup.get(0).rpc.loan.depositToVault({
+        id: vaultId, from: collateralAddress, amount: '10000@DFI'
+      })
+      expect(typeof depositId).toStrictEqual('string')
+      await tGroup.get(0).generate(1)
+      await tGroup.waitForSync()
+
+      const vaultAfter = await tGroup.get(0).container.call('getvault', [vaultId])
+      const vaultAfterDFIAcc = vaultAfter.collateralAmounts.find((amt: string) => amt.split('@')[1] === 'DFI')
+      const vaultAFterDFIAmt = Number(vaultAfterDFIAcc.split('@')[0])
+      expect(vaultAFterDFIAmt - vaultBeforeDFIAmt).toStrictEqual(10000)
```

Comment:
can use `vaultAfter.collateralValue` to compare the same details. But keep it the way it is. And add some more tests with collateralization factor != 1, and there, compare the values by using the prices set by the oracles.

Reply:
added the comparison on setting factor 1 and factor 0.5 on collateral token with calculation
eg:
```ts
// calculate BTC collateral value with factor
const btcDeposit = 1 * 10000 * 0.5 // deposit 1 BTC * priceFeed 10000 USD * 0.5 factor
expect(vaultAfter.collateralValue - vaultBefore.collateralValue).toStrictEqual(btcDeposit)
 ```